### PR TITLE
Bitbucket Server hook support

### LINKED ISSUE #12: Bitbucket Server support
Hey there 👋 

We're using Bitbucket Server and we'd like flux-recv to support it. Bitbucket Server (and probably Cloud too) supports push events webhooks ([see here](https://confluence.atlassian.com/bitbucketserver067/managing-webhooks-in-bitbucket-server-979426856.html)).
I could implement the handler myself once I have time but if anyone wants to jump in, feel free.

### PR DESCRIPTION
**Pull request by abursavich**
I found the Bitbucket Server [webhooks documentation](https://confluence.atlassian.com/bitbucketserver/event-payload-938025882.html) to be very lacking, but I filled in the gaps experimentally.

Fixes #12

### CODE REVIEW

File: bitbucket_server.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		http.Error(w, "Missing repository SSH clone link", http.StatusBadRequest)
+		log(BitbucketServer, "missing repository SSH clone link")
+		return
+	}
+
+	const branchPrefix = "refs/heads/"
+	ctx, cancel := context.WithTimeout(r.Context(), timeout)
+	defer cancel()
+	for _, v := range event.Changes {
+		if !strings.HasPrefix(v.RefID, branchPrefix) {
```
Comment (sa-spag) [inline bitbucket_server.go]: We could check for `type == "BRANCH"` here instead.

Comment (abursavich) [reply bitbucket_server.go]: Done.

File: README.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ exposing to the internet, because
  - `GitHub` push events (and ping events)
  - `DockerHub` image push events
  - `GitLab` push events
+ - `Bitbucket` push events
```
Comment (squaremo) [inline README.md]: Suggestion: mention both bitbucket cloud (which I failed to do in #13) and bitbucket server.

Comment (squaremo) [reply README.md]: I'll do this in a subsequent PR, to save you some faff.

**Pull request merged**